frontend/src/metabase/components/VirtualizedList/VariableHeightVirtualizedList.tsx

```diff
@@ -0,0 +1,68 @@
+import { useVirtualizer } from "@tanstack/react-virtual";
+import { useRef } from "react";
+
+import { Box } from "metabase/ui";
+
+/**
+ * Use this when you need to virtualize a 1-dimensional list of items with variable heights.
+ * You can override the default wrapper element (just a div) by passing a `Wrapper` prop.
+ *
+ * It must be inside an element with a fixed height.
+ */
+export function VirtualizedList({
+  children,
+  Wrapper = Box,
+  estimatedItemSize = 32,
+}: {
+  children: React.ReactNode[];
+  Wrapper?: React.JSXElementConstructor<any>;
+  estimatedItemSize?: number;
+}) {
+  const parentRef = useRef(null);
+
+  const virtualizer = useVirtualizer({
+    count: children.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => estimatedItemSize,
+  });
+
+  const items = virtualizer.getVirtualItems();
+
+  return (
+    <div
+      ref={parentRef}
+      style={{
```

Comment:
can we have styles defined as a css module and not inlined?

Reply:
since the virtualizer has to set some of them programmatically, I thought for this file it would make sense to keep them all inline so you don't have to look elsewhere to figure out what's happening